pkg/task/task.go

```diff
@@ -146,3 +207,52 @@ type CheckpointInfo struct {
 type CompletedTask struct {
 	Checkpoint TaskCheckpoint `json:"checkpoint"`
 }
+
+func newCheckpointInfo() CheckpointInfo {
+	return CheckpointInfo{
+		CompletedTasks: make(map[string]*CompletedTask),
+	}
+}
+
+func (c CheckpointInfo) taskCompleted(name string, completedTask *CompletedTask) {
+	c.CompletedTasks[name] = completedTask
+}
+
+func GetCheckpointFile(file string) (*CheckpointInfo, error) {
+	logger.V(4).Info("Reading checkpoint", "file", file)
+	content, err := os.ReadFile(file)
+	if err != nil {
+		return nil, fmt.Errorf("failed reading checkpoint file: %v\n", err)
+	}
+	checkpointInfo := &CheckpointInfo{}
+	err = yaml.Unmarshal(content, checkpointInfo)
+	if err != nil {
+		return nil, fmt.Errorf("failed unmarshalling checkpoint: %v\n", err)
+	}
+
+	return checkpointInfo, nil
+}
+
+func UnmarshalTaskCheckpoint(taskCheckpoint TaskCheckpoint, config interface{}) error {
+	checkpointYaml, err := yaml.Marshal(taskCheckpoint)
+	if err != nil {
+		return nil
+	}
+	return yaml.Unmarshal(checkpointYaml, config)
+}
+
+func setupCheckpointInfo(commandContext *CommandContext, checkpointFileName string) (CheckpointInfo, error) {
+	checkpointInfo := newCheckpointInfo()
+	checkpointFilePath := filepath.Join(commandContext.Writer.Dir(), "generated", checkpointFileName)
```

Comment:
I don't like hardcoding here `generated`, knowing the dir is a writer's concern
Is there any way to get that info from the writer?

Reply:
Added a new method in the writer to return the temp directory